### PR TITLE
Rename FLOAT type to REAL

### PR DESCRIPTION
**Pull request by maciejgrzybek**
SQL standard defines FLOAT to be parametrized approximate numeric type.
Though section 6.1. of ISO/IES 9075-2:2011 specifies precision of FLOAT as optional.
In the same time it is not defined what precision should be used for FLOAT while no precision is given.
The only constraints for precision provided for REAL, FLOAT and DOUBLE types is (6.1.29-31 of the same document):
- precision of FLOAT is equal to or greater than the value specified (as a parameter)
- precision of REAL is implementation-defined
- precision of DOUBLE is greater than precision of REAL

Thus, for consistency with parameterized DECIMAL which we implemented to satisfy below constraint:
DECIMAL = DECIMAL(MAX_PRECISION)
unparametrized FLOAT should become FLOAT(MAX_PRECISION) which should be equal to DOUBLE.
Wraping up above, 32b floating point number should become distinct from FLOAT type, so it becomes REAL type.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/RealHistogramAggregation.java [hunk 1]
```diff
@@ -64,10 +64,10 @@ public static void output(DoubleHistogramAggregation.State state, BlockBuilder o
         }
         else {
             Map<Double, Double> value = state.get().getBuckets();
-            BlockBuilder blockBuilder = FloatType.FLOAT.createBlockBuilder(new BlockBuilderStatus(), value.size() * 2);
+            BlockBuilder blockBuilder = RealType.REAL.createBlockBuilder(new BlockBuilderStatus(), value.size() * 2);
```
Comment (cawallin) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/RealHistogramAggregation.java]: static import

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/RealSumAggregation.java [hunk 2]
```diff
@@ -58,14 +58,14 @@ public static void combine(NullableDoubleState state, NullableDoubleState otherS
...
-    @OutputFunction(StandardTypes.FLOAT)
+    @OutputFunction(StandardTypes.REAL)
     public static void output(NullableDoubleState state, BlockBuilder out)
     {
         if (state.isNull()) {
             out.appendNull();
         }
         else {
-            FloatType.FLOAT.writeLong(out, floatToRawIntBits((float) state.getDouble()));
+            RealType.REAL.writeLong(out, floatToRawIntBits((float) state.getDouble()));
```
Comment (cawallin) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/RealSumAggregation.java]: static import

Comment (maciejgrzybek) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/RealSumAggregation.java]: Isn't it unrelated change? :P I will do, sure. Thanks for thorough read. :)

Comment (cawallin) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/RealSumAggregation.java]: I know, it was nitpicky... but better to get it right now

**Comment by cawallin** [comment]
looks good % making sure tests pass

**Comment by maciejgrzybek** [comment]
@cawallin static imports fixed in 96cca20

**Comment by maciejgrzybek** [comment]
Closing to start PR against prestodb/presto.

**Pull request closed without merging**